### DISCUSSION TITLE
Excalidraw integration

### DISCUSSION THREAD

**Discussion by lpadula** [outline/outline#6346] [Ideas]
Hi! I am a passionate user of both Outline (your fantastic wiki tool) and Excalidraw, the open-source, diagramming tool. However, I've noticed that there isn't a seamless way to integrate the visual diagrams created in Excalidraw with the structured knowledge base in Outline.

This feature would allow users to embed Excalidraw diagrams directly into their Outline wiki pages. Ideally, this would work similarly to how images or other media can be embedded now, but it would provide the dynamic capability to edit the diagrams within Outline through a direct link to Excalidraw.

**Comment by helmut72** [comment]
Would be cool. If this happens, please also support self-hosted Excalidraw installations.

**Comment by helmut72** [comment]
In the meantime Nextcloud has added Excalidraw:
https://github.com/nextcloud/whiteboard

... same for Docmost:
https://github.com/docmost/docmost/releases/tag/v0.3.0

**Comment by caffeinewriter** [comment]
I just ran into this today. I embedded it using a plain `/embed`, but it's not quite as smooth as I'd like. I used a read-only link, but reloading causes it to warn that it's going to replace the content that's there, and it starts out quite zoomed in, rather than showing an overview of the whole diagram. 

Excalidraw seems to have some [fairly robust integration support](https://docs.excalidraw.com/docs/@excalidraw/excalidraw/integration).

Example of behavior when coming back to the page:
<img width="643" alt="Screenshot 2025-01-30 at 11 14 07 AM" src="https://github.com/user-attachments/assets/0af8c1d7-c120-458e-b85e-4f7ed8048e31" />

**Comment by krmozejko** [comment]
That would be a great addition. We have recently started using outline as our main documentation tool in the company and we are missing support for excalidraw integration.

**Comment by seamon67** [comment]
There is a way to this in outline right now.
This requires excalidraw to be setup in a very specific way.
You need 4 containers:
1. Iframe Excalidraw: https://github.com/jhermen-List/excalidraw-iframe
You have to build the image using the dynamic env Dockerfile.
2. Excalidraw Storage: https://github.com/alswl/excalidraw-storage-backend
3. Excalidraw Room: https://hub.docker.com/r/excalidraw/excalidraw-room
4. Nginx: to combine all 3 of these together.
5. Also, a DB: MongoDB works well.

You can now post the room url from excalidraw into an Outline embed and it will be persistent with collaboration directly inside Outline.

**Comment by Rodaviva29** [comment]
This would be lovely or a own whiteboard feat.

**Comment by danielraab** [comment]
I would like to see this, like docmost implemented it.

**Reply by Darkdragon84** [reply]
Docmost has a nice integration, but it's still not collaborative. When you save and exit you overwrite whatever changes someone else made in the meantime. A full live session integration would achieve this I suppose.

**Comment by narthanaj** [comment]
is this ongoing :D

**Reply by zguiyang** [reply]
+1

**Comment by helmut72** [comment]
There is a PR from October 2025, but wasn't accepted - too much code.
https://github.com/outline/outline/pull/10349

**Comment by oakbrad** [comment]
The big benefit of Excalidraw is that like Outline, it supports multi-player collaboration. The current options (like diagrams.net) do not.

**Comment by Raphmatt** [comment]
And many users love it. There is an [Excalidraw plugin](https://github.com/zsviczian/obsidian-excalidraw-plugin) for Obsidian, and it's the most [downloaded plugin](https://www.obsidianstats.com/) in the entire plugin marketplace of obsidian. (with over 5 Million downloads!)

**Comment by saschabrockel** [comment]
For me, Outline is the best wiki overall. That is the only thing that is really missing. Docmost has it, and we had a customer who chose Docmost just because of its integration. It would be lovely if Excalidraw were integrated.

**Reply by Rodaviva29** [reply]
I hope someday we get this.

**Comment by mafloh** [comment]
+1

**Comment by helmut72** [comment]
Maybe Excalidash could be a solution in future? Right now it's in beta:
https://github.com/ZimengXiong/ExcaliDash

I miss file managment on the base Excalidraw website and base Docker image. Excalidash brings file management, collaboration and also authentication via OIDC.

Then Outline is for documents, Grist for tables and Excalidash for drawings.

**Reply by thelazyoxymoron** [reply]
Excalidash looks exactly what I was looking for. Thank you.

**Comment by tovoro** [comment]
+1

**Comment by jeremymoll** [comment]
Adding my voice to this discussion after coming here to submit a feature request.

I would really love to see **Excalidraw** in Outline!

**Comment by asupersheep** [comment]
+1
